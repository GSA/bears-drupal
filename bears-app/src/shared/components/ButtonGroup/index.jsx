import { Button } from '../index'
import PropTypes from 'prop-types'

/**
 * a functional component to create a group of navigational buttons.
 * @function
 * @param {object} buttonOne - The object config for the first button element
 * @param {object} buttonTwo - The object config for the second button element
 * @return {html} returns a unorderd list of buttons
 */
const ButtonGroup = ({ buttonOne, buttonTwo }) => {
  return (
    <ul className="usa-button-group width-full">
      <li className="usa-button-group__item width-full">
        <Button {...buttonOne}>{buttonOne && buttonOne.children}</Button>
      </li>
      <li className="usa-button-group__item width-full">
        <Button {...buttonTwo}>{buttonOne && buttonTwo.children}</Button>
      </li>
    </ul>
  )
}

ButtonGroup.propTypes = {
  buttonOne: PropTypes.object,
  buttonTwo: PropTypes.object,
}

export default ButtonGroup
