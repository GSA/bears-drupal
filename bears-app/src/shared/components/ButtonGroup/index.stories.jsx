import ButtonGroup from './index.jsx'

export default {
  component: ButtonGroup,
  tags: ['autodocs'],
  args: {
    buttonOne: {
      secondary: true,
      children: 'Back',
      onClick: () => alert('navigate back'),
    },
    buttonTwo: {
      children: 'Continue',
      onClick: () => alert('navigate forward'),
    },
  },
}

export const Primary = {}
export const Modal = {
  args: {
    ...Primary.args,
    buttonOne: {
      children: 'Review Your Selection',
      onClick: () => alert('navigate back'),
    },
    buttonTwo: {
      children: 'See Results',
      onClick: () => alert('navigate forward'),
    },
  },
}
