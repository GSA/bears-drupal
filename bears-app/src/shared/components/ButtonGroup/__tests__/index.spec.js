import renderer from 'react-test-renderer'
import ButtonGroup from '../index.jsx'

describe('ButtonGroup', () => {
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(<ButtonGroup />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
