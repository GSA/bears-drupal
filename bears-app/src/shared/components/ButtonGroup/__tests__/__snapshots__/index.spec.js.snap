// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup renders a match to the previous snapshot 1`] = `
<ul
  className="usa-button-group"
>
  <li
    className="usa-button-group__item"
  >
    <button
      className=""
      type="button"
    />
  </li>
  <li
    className="usa-button-group__item"
  >
    <button
      className=""
      type="button"
    />
  </li>
</ul>
`;
